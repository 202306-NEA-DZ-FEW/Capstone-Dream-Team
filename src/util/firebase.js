// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getAuth } from "firebase/auth";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: "capstone-project-944c9.firebaseapp.com",
    projectId: "capstone-project-944c9",
    storageBucket: "capstone-project-944c9.appspot.com",
    messagingSenderId: "455661891383",
    appId: "1:455661891383:web:65bbc812478ffda076b4b2",
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
