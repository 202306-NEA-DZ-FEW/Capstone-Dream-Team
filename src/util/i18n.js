import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
