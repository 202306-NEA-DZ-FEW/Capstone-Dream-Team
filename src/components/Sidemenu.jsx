import { onAuthStateChanged, signOut } from "firebase/auth";
import { doc, getDoc } from "firebase/firestore";
import Image from "next/image";
import Link from "next/link";
import { useRouter } from "next/router";
import { useTranslation } from "next-i18next";
import React from "react";
import { useEffect, useState } from "react";
import { AiOutlineHome, AiOutlineSetting } from "react-icons/ai";
import { HiOutlineChartBarSquare } from "react-icons/hi2";
import { IoIosArrowDropleft, IoIosArrowDropright } from "react-icons/io";
import { IoLogOutOutline } from "react-icons/io5";
import { IoEarthOutline } from "react-icons/io5";
import { LiaHistorySolid } from "react-icons/lia";

import AddMeals from "./Meals/addMeals";
import Adminhistory from "./History/Adminhistory";
import Overview from "./Overview/overview";
import Settings from "./Settings";
import { auth, db } from "../util/firebase";

export default function Sidemenu(props) {
    const { t } = useTranslation("common");
    const components = [
        {
            name: "sidemenu.overview",
            icon: <AiOutlineHome size={20}></AiOutlineHome>,
            element: <Overview />,
        },
        {
            name: "sidemenu.Meals",
            icon: <HiOutlineChartBarSquare size={20} />,
            element: <AddMeals />,
        },
        {
            name: "sidemenu.History",
            icon: <LiaHistorySolid size={20} />,
            element: <Adminhistory />,
        },
        {
            name: "sidemenu.Settings",
            icon: <AiOutlineSetting size={20} />,
            element: <Settings />,
        },
    ];
    // Import and destructure the 'useTheme' hook and React state management functions.

    // Initialize state variables for component mounting, current theme, user data, and authenticated user.

    const [userData, setUserData] = useState(null); // State to store user data.
    const [authUser, setAuthUser] = useState(null); // State to store the authenticated user.
    const [open, setOpen] = useState(false);
    // Use 'useEffect' to run code when 'authUser' changes.
    const [clicked, setClicked] = useState(false);
    const [mousenter, setMousenter] = useState(false);
    const router = useRouter();

    // Function to change the language
    const changeLanguage = (newLanguage) => {
        const { pathname, query, asPath } = router;

        // Use the router to change the locale in the URL
        router.push({ pathname, query }, asPath, { locale: newLanguage });
    };

    function handleClick() {
        setClicked(!clicked);
    }
    useEffect(() => {
        // Use Firebase's 'onAuthStateChanged' to listen for changes in user authentication state.
        onAuthStateChanged(auth, (user) => {
            // If a user is authenticated, set 'authUser' to the user; otherwise, set it to null.
            user ? setAuthUser(user) : setAuthUser(null);
        });

        // Define a function to fetch user information from Firestore.
        const fetchInformation = async () => {
            if (authUser) {
                // Check if an authenticated user exists.
                const userId = authUser.uid; // Get the user's unique identifier (UID).
                const docRef = doc(db, "restaurant", userId); // Create a reference to the user's Firestore document.
                const docSnap = await getDoc(docRef); // Fetch the user's document.

                if (docSnap.exists()) {
                    // Check if the document exists.
                    // Extract and store user data, including the user's ID.
                    const userData = { ...docSnap.data(), id: userId };
                    setUserData(userData);
                }
            }
        };

        // Call the 'fetchInformation' function to initiate data fetching.
        fetchInformation();
    }, [authUser]); // Run the effect when 'authUser' changes.

    // Define a function to handle user logout.
    const handleLogout = async () => {
        // Sign out the authenticated user using Firebase's 'signOut' function.
        await signOut(auth);
    };
    const handleShow = () => {
        setOpen(!open);
    };

    return (
        <>
            <nav
                style={
                    router.locale === "ar"
                        ? {
                              backgroundPositionX: "right",
                              backgroundImage:
                                  "url(/images/sidemenu/background.png)",
                          }
                        : {
                              backgroundImage:
                                  "url(images/sidemenu/background.png)",
                          }
                }
                className={`col-span-1 z-50 h-screen fixed duration-300 bg-cover ${
                    open ? " w-60" : "w-20 md:w-60"
                }`}
                onMouseLeave={() => setOpen(false)}
            >
                {router.locale === "ar" ? (
                    <IoIosArrowDropleft
                        onClick={handleShow}
                        className={`z-40 md:hidden absolute top-16 -left-6 cursor-pointer w-8 h-10 duration-300 ${
                            open && "rotate-180 "
                        } `}
                    ></IoIosArrowDropleft>
                ) : (
                    <IoIosArrowDropright
                        onClick={handleShow}
                        className={`z-40 md:hidden absolute top-16 -right-4 cursor-pointer w-8 h-10 duration-300 ${
                            open && "rotate-180 "
                        } `}
                    ></IoIosArrowDropright>
                )}
                <div className=' w-full h-full pl-3 shadow-xl border-r border-gray-200 flex flex-col justify-between items-start  gap-8'>
                    <div className='w-full flex-col justify-start items-start gap-2 flex '>
                        <div className=' mx-auto md:border-b md:border-gray-400 items-center gap-3 my-6 md:mx-auto justify-start md:justify-center '>
                            <div
                                className={`w-16 h-16 bg-white rounded-full border border-blue-200 md:w-20 mt-2 mb-4 md:mb-6  md:h-20 relative  ease-out duration-500  ${
                                    !open &&
                                    (router.locale === "ar"
                                        ? "ml-32 scale-50 p-0 mb-2 md:transform-none md:mx-auto md:mt-2 md:mb-6"
                                        : "mr-24 scale-50 p-0 mb-2 md:transform-none md:mx-auto md:mt-2 md:mb-6")
                                }$`}
                            >
                                <Link href='/' className=''>
                                    <img
                                        src='/images/home/Navbar/sidebarLogo.png'
                                        alt='logo'
                                    ></img>
                                </Link>
                            </div>
                        </div>
                        <div className='w-full md:justify-center items-center gap-3  inline-flex justify-start lg:inline-flex'>
                            <Image
                                className={`w-12 h-12 relative rounded-2xl ease-out duration-700 ${
                                    open && "rotate-[360deg]"
                                }`}
                                src={
                                    userData && userData.image
                                        ? userData.image
                                        : "/images/placeholderImage.png"
                                }
                                width={40}
                                height={40}
                                alt='admin photo'
                            />

                            <div
                                className={`flex-col justify-center items-start inline-flex ${
                                    !open && "scale-0 md:transform-none"
                                }`}
                            >
                                <div className="w-[150px] break-words text-zinc-950 dark:text-white text-base font-bold font-['Open Sans']">
                                    {userData
                                        ? userData.restaurantName
                                        : "loading..."}
                                </div>
                                <div className="w-[150px] break-all text-neutral-800  dark:text-white text-[10px] lg:text-[14px] font-normal font-['Open Sans']">
                                    {userData ? userData.email : "loading..."}
                                </div>
                            </div>
                        </div>

                        <div className='w-full flex-col pt-4 pb-2 justify-start items-start gap-4 flex'>
                            {components.map((component, index) => (
                                <div
                                    key={index}
                                    className=' flex-col justify-start items-start gap-1 flex '
                                >
                                    <div
                                        className={` h-4 p-4 rounded-lg justify-start items-center gap-2 inline-flex hover:text-white hover:bg-blue-400 active:bg-blue-700 focus:outline-none focus:bg-blue-600 ${
                                            clicked === true
                                                ? "focus:outline-none focus-within:text-orange-600 focus-current:border-b-2 "
                                                : ""
                                        }`}
                                        onClick={() => {
                                            props.handleClick(
                                                component.element
                                            );
                                            setClicked(true);
                                        }}
                                    >
                                        <div className='relative focus:outline-none'>
                                            {component.icon}
                                        </div>
                                        <button
                                            className={` focus:outline-none text-[10px] md:text-sm font-normal font-['Open Sans']  duration-300 ${
                                                !open &&
                                                "scale-0 md:transform-none"
                                            }`}
                                        >
                                            {t(`${component.name}`)}
                                        </button>
                                    </div>
                                </div>
                            ))}
                        </div>
                    </div>
                    <div className=' w-full self-stretch h-full flex-col justify-start pt-1 items-start  flex'>
                        <div className='w-full flex-col justify-start items-start gap-1 flex'>
                            <div
                                className='h-4 p-4 rounded-lg justify-start items-center gap-2 inline-flex cursor-pointer  hover:text-white hover:bg-blue-400 active:bg-blue-700 focus:outline-none focus:bg-blue-600'
                                onMouseEnter={() => setMousenter(true)}
                                onMouseLeave={() => setMousenter(false)}
                            >
                                <div className='focus:outline-none relative'>
                                    <IoEarthOutline size={20}></IoEarthOutline>{" "}
                                </div>
                                <button
                                    className={`text-[#333333] hover:text-black  dark:text-white text-[10px] md:text-sm font-normal font-['Open Sans'] leading-snug focus:outline-none duration-300 ${
                                        !open && "scale-0 md:transform-none"
                                    }`}
                                >
                                    {t("sidemenu.language")}
                                </button>
                                <div className='relative'>
                                    {mousenter === true && (
                                        <div
                                            className={` relative  ${
                                                props.locale === "en"
                                                    ? "absolute py-1 w-24 h-16 bg-blue-200 border rounded-xl shadow-xl "
                                                    : "relative  py-1 w-24 h-16 bg-blue-200 border rounded-xl shadow-xl "
                                            }`}
                                        >
                                            <ul>
                                                <li
                                                    className={`${
                                                        props.locale === "en"
                                                            ? "block w-full text-center mx-auto py-1 text-sm capitalize text-gray-800 hover:bg-orange-600 hover:text-white"
                                                            : "block w-full text-center mx-auto py-1 text-sm capitalize text-gray-800 hover:bg-orange-600 hover:text-white"
                                                    }`}
                                                >
                                                    {" "}
                                                    <button
                                                        onClick={() =>
                                                            changeLanguage("en")
                                                        }
                                                    >
                                                        English
                                                    </button>
                                                </li>
                                                <li
                                                    className={`${
                                                        props.locale === "en"
                                                            ? "block w-full text-center mx-auto py-1 text-sm capitalize text-gray-800 hover:bg-orange-600 hover:text-white"
                                                            : "block w-full text-center mx-auto py-1 text-sm capitalize text-gray-800 hover:bg-orange-600 hover:text-white"
                                                    }`}
                                                >
                                                    {" "}
                                                    <button
                                                        onClick={() =>
                                                            changeLanguage("ar")
                                                        }
                                                    >
                                                        العربية
                                                    </button>
                                                </li>
                                            </ul>
                                        </div>
                                    )}
                                </div>
                            </div>
                            <Link href='/'>
                                <div
                                    className='p-4 rounded-lg justify-start items-center gap-1 inline-flex hover:text-white hover:bg-blue-400 active:bg-blue-700 focus:bg-blue-600'
                                    onClick={handleLogout}
                                >
                                    <div className=' w-6 h-4 relative'>
                                        <IoLogOutOutline size={20} />
                                    </div>
                                    <button className="text-[#333333]  dark:text-white text-[10px] md:text-sm font-normal font-['Open Sans'] leading-snug">
                                        <div
                                            className={`duration-300 ${
                                                !open &&
                                                "scale-0  md:transform-none"
                                            }`}
                                        >
                                            {t("sidemenu.Logout")}
                                        </div>
                                    </button>
                                </div>{" "}
                            </Link>
                        </div>
                    </div>
                </div>
            </nav>
        </>
    );
}
