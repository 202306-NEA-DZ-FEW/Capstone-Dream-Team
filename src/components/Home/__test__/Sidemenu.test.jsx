import renderer from "react-test-renderer";
import { onAuthStateChanged, signOut } from "firebase/auth"; // Import Firebase functions
import { getAuth } from "firebase/auth"; // Import getAuth function

jest.mock("firebase/auth", () => ({
    getAuth: jest.fn(), // Mock getAuth function
    onAuthStateChanged: jest.fn(),
    signOut: jest.fn(),
}));

import { I18nextProvider } from "react-i18next";
import i18n from "@/util/i18n";
import Sidemenu from "../../Sidemenu";

jest.mock("next/router", () => ({
    useRouter: jest.fn().mockReturnValue({
        push: jest.fn(),
    }),
}));

it("renders correctly", () => {
    // Mock the usage of getAuth if it's being used within Sidemenu component
    // getAuth.mockReturnValue(...);

    const tree = renderer
        .create(
            <I18nextProvider i18n={i18n}>
                <Sidemenu language='en' />
            </I18nextProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
