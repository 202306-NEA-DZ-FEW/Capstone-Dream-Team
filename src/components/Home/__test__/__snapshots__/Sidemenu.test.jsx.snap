// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="col-span-1 z-50 h-screen fixed duration-300 bg-cover w-20 md:w-60"
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(images/sidemenu/background.png)",
    }
  }
>
  <svg
    className="z-40 md:hidden absolute top-16 -right-4 cursor-pointer w-8 h-10 duration-300 false "
    fill="currentColor"
    height="1em"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M210.7 147.6c7.5-7.5 19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81.1-81.9-81.1c-7.6-7.4-7.6-19.6 0-27.2z"
    />
    <path
      d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm32 0c0-47 18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256z"
    />
  </svg>
  <div
    className=" w-full h-full pl-3 shadow-xl border-r border-gray-200 flex flex-col justify-between items-start  gap-8"
  >
    <div
      className="w-full flex-col justify-start items-start gap-2 flex "
    >
      <div
        className=" mx-auto md:border-b md:border-gray-400 items-center gap-3 my-6 md:mx-auto justify-start md:justify-center "
      >
        <div
          className="w-16 h-16 bg-white rounded-full border border-blue-200 md:w-20 mt-2 mb-4 md:mb-6  md:h-20 relative  ease-out duration-500  mr-24 scale-50 p-0 mb-2 md:transform-none md:mx-auto md:mt-2 md:mb-6$"
        >
          <a
            className=""
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <img
              alt="logo"
              src="/images/home/Navbar/sidebarLogo.png"
            />
          </a>
        </div>
      </div>
      <div
        className="w-full md:justify-center items-center gap-3  inline-flex justify-start lg:inline-flex"
      >
        <img
          alt="admin photo"
          className="w-12 h-12 relative rounded-2xl ease-out duration-700 false"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimages%2FplaceholderImage.png&w=96&q=75"
          srcSet="/_next/image?url=%2Fimages%2FplaceholderImage.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2FplaceholderImage.png&w=96&q=75 2x"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={40}
        />
        <div
          className="flex-col justify-center items-start inline-flex scale-0 md:transform-none"
        >
          <div
            className="w-[150px] break-words text-zinc-950 dark:text-white text-base font-bold font-['Open Sans']"
          >
            loading...
          </div>
          <div
            className="w-[150px] break-all text-neutral-800  dark:text-white text-[10px] lg:text-[14px] font-normal font-['Open Sans']"
          >
            loading...
          </div>
        </div>
      </div>
      <div
        className="w-full flex-col pt-4 pb-2 justify-start items-start gap-4 flex"
      >
        <div
          className="w-4/5 h-4 p-4 rounded-lg justify-start items-center gap-2 inline-flex  "
          onClick={[Function]}
        >
          <div
            className="relative focus:outline-none"
          >
            <svg
              fill="currentColor"
              height={20}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </div>
          <button
            className=" focus:outline-none text-[10px] md:text-sm font-normal font-['Open Sans']  duration-300 scale-0 md:transform-none"
          >
            sidemenu.overview
          </button>
        </div>
        <div
          className="w-4/5 h-4 p-4 rounded-lg justify-start items-center gap-2 inline-flex  "
          onClick={[Function]}
        >
          <div
            className="relative focus:outline-none"
          >
            <svg
              aria-hidden="true"
              fill="none"
              height={20}
              stroke="currentColor"
              strokeWidth="1.5"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.5 14.25v2.25m3-4.5v4.5m3-6.75v6.75m3-9v9M6 20.25h12A2.25 2.25 0 0020.25 18V6A2.25 2.25 0 0018 3.75H6A2.25 2.25 0 003.75 6v12A2.25 2.25 0 006 20.25z"
                strokeLinecap="round"
                strokeLinejoin="round"
              />
            </svg>
          </div>
          <button
            className=" focus:outline-none text-[10px] md:text-sm font-normal font-['Open Sans']  duration-300 scale-0 md:transform-none"
          >
            sidemenu.Meals
          </button>
        </div>
        <div
          className="w-4/5 h-4 p-4 rounded-lg justify-start items-center gap-2 inline-flex  "
          onClick={[Function]}
        >
          <div
            className="relative focus:outline-none"
          >
            <svg
              fill="currentColor"
              height={20}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 16 4 C 11.832031 4 8.152344 6.113281 6 9.34375 L 6 6 L 4 6 L 4 13 L 11 13 L 11 11 L 7.375 11 C 9.101563 8.019531 12.296875 6 16 6 C 21.535156 6 26 10.464844 26 16 C 26 21.535156 21.535156 26 16 26 C 10.464844 26 6 21.535156 6 16 L 4 16 C 4 22.617188 9.382813 28 16 28 C 22.617188 28 28 22.617188 28 16 C 28 9.382813 22.617188 4 16 4 Z M 15 8 L 15 17 L 22 17 L 22 15 L 17 15 L 17 8 Z"
              />
            </svg>
          </div>
          <button
            className=" focus:outline-none text-[10px] md:text-sm font-normal font-['Open Sans']  duration-300 scale-0 md:transform-none"
          >
            sidemenu.History
          </button>
        </div>
        <div
          className="w-4/5 h-4 p-4 rounded-lg justify-start items-center gap-2 inline-flex  "
          onClick={[Function]}
        >
          <div
            className="relative focus:outline-none"
          >
            <svg
              fill="currentColor"
              height={20}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a443.74 443.74 0 0 0-79.7-137.9l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.4a351.86 351.86 0 0 0-99 57.4l-81.9-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a446.02 446.02 0 0 0-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0 0 25.8 25.7l2.7.5a449.4 449.4 0 0 0 159 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-85a350 350 0 0 0 99.7-57.6l81.3 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 0 1-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 0 1-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 0 1 512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 0 1 400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 0 1 624 502c0 29.9-11.7 58-32.8 79.2z"
              />
            </svg>
          </div>
          <button
            className=" focus:outline-none text-[10px] md:text-sm font-normal font-['Open Sans']  duration-300 scale-0 md:transform-none"
          >
            sidemenu.Settings
          </button>
        </div>
      </div>
    </div>
    <div
      className=" w-full self-stretch h-full flex-col justify-start pt-1 items-start  flex"
    >
      <div
        className="w-full flex-col justify-start items-start gap-1 flex"
      >
        <div
          className="h-4 p-4 rounded-lg justify-start items-center gap-2 inline-flex cursor-pointer  focus:outline-none focus:text-orange-600"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="focus:outline-none relative"
          >
            <svg
              fill="currentColor"
              height={20}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 256c0-114.87-93.13-208-208-208S48 141.13 48 256s93.13 208 208 208 208-93.13 208-208z"
                fill="none"
                strokeMiterlimit="10"
                strokeWidth="32"
              />
              <path
                d="M445.57 172.14c-16.06.1-14.48 29.73-34.49 15.75-7.43-5.18-12-12.71-21.33-15-8.15-2-16.5.08-24.55 1.47-9.15 1.58-20 2.29-26.94 9.22-6.71 6.67-10.26 15.62-17.4 22.33-13.81 13-19.64 27.19-10.7 45.57 8.6 17.67 26.59 27.26 46 26 19.07-1.27 38.88-12.33 38.33 15.38-.2 9.8 1.85 16.6 4.86 25.71 2.79 8.4 2.6 16.54 3.24 25.21 1.18 16.2 4.16 34.36 12.2 48.67l15-21.16c1.85-2.62 5.72-6.29 6.64-9.38 1.63-5.47-1.58-14.87-1.95-21s-.19-12.34-1.13-18.47c-1.32-8.59-6.4-16.64-7.1-25.13-1.29-15.81 1.6-28.43-10.58-41.65-11.76-12.75-29-15.81-45.47-13.22-8.3 1.3-41.71 6.64-28.3-12.33 2.65-3.73 7.28-6.79 10.26-10.34 2.59-3.09 4.84-8.77 7.88-11.18s17-5.18 21-3.95 8.17 7 11.64 9.56a49.89 49.89 0 0021.81 9.36c13.66 2 42.22-5.94 42-23.46-.04-8.4-7.84-20.1-10.92-27.96zM287.45 316.3c-5.33-22.44-35.82-29.94-52.26-42.11-9.45-7-17.86-17.81-30.27-18.69-5.72-.41-10.51.83-16.18-.64-5.2-1.34-9.28-4.14-14.82-3.41-10.35 1.36-16.88 12.42-28 10.92-10.55-1.42-21.42-13.76-23.82-23.81-3.08-12.92 7.14-17.11 18.09-18.26 4.57-.48 9.7-1 14.09.67 5.78 2.15 8.51 7.81 13.7 10.67 9.73 5.33 11.7-3.19 10.21-11.83-2.23-12.94-4.83-18.22 6.71-27.12 8-6.14 14.84-10.58 13.56-21.61-.76-6.48-4.31-9.41-1-15.86 2.51-4.91 9.4-9.34 13.89-12.27 11.59-7.56 49.65-7 34.1-28.16-4.57-6.21-13-17.31-21-18.83-10-1.89-14.44 9.27-21.41 14.19-7.2 5.09-21.22 10.87-28.43 3-9.7-10.59 6.43-14.07 10-21.46s-8.27-21.36-14.61-24.9l-29.81 33.43a41.52 41.52 0 008.34 31.86c5.93 7.63 15.37 10.08 15.8 20.5.42 10-1.14 15.12-7.68 22.15-2.83 3-4.83 7.26-7.71 10.07-3.53 3.43-2.22 2.38-7.73 3.32-10.36 1.75-19.18 4.45-29.19 7.21C95.34 199.94 93.8 172.69 86.2 162l-25 20.19c-.27 3.31 4.1 9.4 5.29 13 6.83 20.57 20.61 36.48 29.51 56.16 9.37 20.84 34.53 15.06 45.64 33.32 9.86 16.2-.67 36.71 6.71 53.67 5.36 12.31 18 15 26.72 24 8.91 9.09 8.72 21.53 10.08 33.36a305.22 305.22 0 007.45 41.28c1.21 4.69 2.32 10.89 5.53 14.76 2.2 2.66 9.75 4.95 6.7 5.83 4.26.7 11.85 4.68 15.4 1.76 4.68-3.84 3.43-15.66 4.24-21 2.43-15.9 10.39-31.45 21.13-43.35 10.61-11.74 25.15-19.69 34.11-33 8.73-12.98 11.36-30.49 7.74-45.68zm-33.39 26.32c-6 10.71-19.36 17.88-27.95 26.39-2.33 2.31-7.29 10.31-10.21 8.58-2.09-1.24-2.8-11.62-3.57-14a61.17 61.17 0 00-21.71-29.95c-3.13-2.37-10.89-5.45-12.68-8.7-2-3.53-.2-11.86-.13-15.7.11-5.6-2.44-14.91-1.06-20 1.6-5.87-1.48-2.33 3.77-3.49 2.77-.62 14.21 1.39 17.66 2.11 5.48 1.14 8.5 4.55 12.82 8 11.36 9.11 23.87 16.16 36.6 23.14 9.86 5.46 12.76 12.37 6.46 23.62zm-69.6-275.53c4.74 4.63 9.2 10.11 16.27 10.57 6.69.45 13-3.17 18.84 1.38 6.48 5 11.15 11.33 19.75 12.89 8.32 1.51 17.13-3.35 19.19-11.86 2-8.11-2.31-16.93-2.57-25.07 0-1.13.61-6.15-.17-7-.58-.64-5.42.08-6.16.1q-8.13.24-16.22 1.12a207.1 207.1 0 00-57.18 14.65c2.43 1.68 5.48 2.35 8.25 3.22zm171.94 56.18c8.49 0 17.11-3.8 14.37-13.62-2.3-8.23-6.22-17.16-15.76-12.72-6.07 2.82-14.67 10-15.38 17.12-.81 8.08 11.11 9.22 16.77 9.22zm-6.78 42.97c8.67 5.19 21.53 2.75 28.07-4.66 5.11-5.8 8.12-15.87 17.31-15.86a15.4 15.4 0 0110.82 4.41c3.8 3.93 3.05 7.62 3.86 12.54 1.81 11.05 13.66.63 16.75-3.65 2-2.79 4.71-6.93 3.8-10.56-.84-3.39-4.8-7-6.56-10.11-5.14-9-9.37-19.47-17.07-26.74-7.41-7-16.52-6.19-23.55 1.08-5.76 6-12.45 10.75-16.39 18.05-2.78 5.13-5.91 7.58-11.54 8.91-3.1.73-6.64 1-9.24 3.08-7.24 5.7-3.12 19.39 3.74 23.51z"
              />
            </svg>
             
          </div>
          <button
            className="text-[#333333] text-[10px] md:text-sm font-normal font-['Open Sans'] leading-snug focus:outline-none duration-300 scale-0 md:transform-none"
          >
            sidemenu.language
          </button>
          <div
            className="relative"
          />
        </div>
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="p-4 rounded-lg justify-start items-center gap-1 inline-flex hover:text-white hover:bg-blue-400 active:bg-blue-700 focus:bg-blue-600"
            onClick={[Function]}
          >
            <div
              className=" w-6 h-4 relative"
            >
              <svg
                fill="currentColor"
                height={20}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M304 336v40a40 40 0 01-40 40H104a40 40 0 01-40-40V136a40 40 0 0140-40h152c22.09 0 48 17.91 48 40v40m64 160l80-80-80-80m-192 80h256"
                  fill="none"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="32"
                />
              </svg>
            </div>
            <button
              className="text-[#333333]  dark:text-white text-[10px] md:text-sm font-normal font-['Open Sans'] leading-snug"
            >
              <div
                className="duration-300 scale-0  md:transform-none"
              >
                sidemenu.Logout
              </div>
            </button>
          </div>
           
        </a>
      </div>
    </div>
  </div>
</nav>
`;
